src/js/player.js

```diff
@@ -2426,23 +2554,24 @@ class Player extends Component {
     }
 
     // intial sources
-    this.cache_.sources = sources;
     this.changingSrc_ = true;
 
-    // intial source
-    this.cache_.source = sources[0];
-
     // middlewareSource is the source after it has been changed by middleware
     middleware.setSource(this, sources[0], (middlewareSource, mws) => {
       this.middleware_ = mws;
 
+      this.cache_.sources = sources;
+      this.updateSourceCaches_(middlewareSource);
       const err = this.src_(middlewareSource);
 
       if (err) {
         if (sources.length > 1) {
           return this.src(sources.slice(1));
         }
 
+        this.updateSourceCaches_();
+        this.changingSrc_ = false;
```

Comment:
if there was an error and there are no more sources, we should empty the cache and set changingSrc_ to false

Reply:
I guess this makes sense.

Reply:
Looks like the video element doesn't update `video.currentSrc` when a new `src` fails to load properly.

Reply:
yeah `video.currentSrc` remains the previous value but `src` will obviously have changed. I guess we should change the `src` cache but leave the `currentSource` cache on the previous value here?